src/InvariantTheory/invariant_rings.jl

```diff
@@ -15,6 +14,46 @@ group(I::InvRing) = I.group
 
 is_modular(I::InvRing) = I.modular
 
+function _internal_polynomial_ring(I::InvRing)
+  if isdefined(I, :poly_ring_internal)
+    return I.poly_ring_internal
+  end
+  return I.poly_ring
+end
+
+# Return f as an element of R. Assumes that ngens(R) == ngens(parent(f))
+# and coefficient_ring(R) === coefficient_ring(parent(f)). This is not checked.
+function __cast_forced(R::MPolyRing{T}, f::MPolyRingElem{T}) where T
+  F = MPolyBuildCtx(R)
+  for (c, e) in zip(AbstractAlgebra.coefficients(f), AbstractAlgebra.exponent_vectors(f))
+    push_term!(F, c, e)
+  end
+  return finish(F)
```

Comment:
Is there a faster way to cast a polynomial in another polynomial ring that is "the same" apart from the ordering?

Reply:
No